Sources/Caching/DeviceCache.swift

```diff
@@ -338,6 +338,21 @@ class DeviceCache {
         return self.userDefaults.read(Self.productEntitlementMapping)
     }
 
+    // MARK: - SK2
+    func cachedSyncedSK2TransactionIDs(appUserID: String) -> [UInt64]? {
+        return self.userDefaults.read { userDefaults in
+            (userDefaults.array(forKey: CacheKey.syncedSK2TransactionIDs(appUserID).rawValue) as? [NSNumber])?.map {
+                $0.uint64Value
+            }
+        }
+    }
+
+    func cacheSyncedSK2TransactionIDs(syncedSK2TransactionIDs: [UInt64], appUserID: String) {
+        self.userDefaults.write {
+            $0.set(syncedSK2TransactionIDs, forKey: CacheKey.syncedSK2TransactionIDs(appUserID))
+        }
+    }
```

Comment:
Since we're only checking for the most recent transaction now, we store a single value instead of an array?

Reply:
I left a comment in another place, but it being only the most recent transaction scares me a bit since a single transaction can disappear from history if: 
- consumable transaction is finished
- family sharing access is revoked

Reply:
@MarkVillacampa and I decided via DM to keep the caching to all of the IDs via an array. Going to resolve this conversation and continue the discussion about consumables/family sharing to the other thread!